server/src/domain/types.rs

```diff
@@ -167,6 +167,82 @@ impl ValueType for UserId {
     }
 }
 
+#[derive(PartialEq, Eq, Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(from = "String")]
+pub struct DisplayName(String);
+
+impl DisplayName {
+    pub fn new(display_name: &str) -> Self {
+        Self(display_name.to_string())
+    }
+
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+
+    pub fn into_string(self) -> String {
+        self.0
+    }
+}
+
+impl std::fmt::Display for DisplayName {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl From<String> for DisplayName {
+    fn from(s: String) -> Self {
+        Self::new(&s)
```

Comment:
You're making an extra copy here: since you have a `String`, you can just write `Self(s)`